src/registrar/management/commands/utility/terminal_helper.py

```diff
@@ -58,6 +59,54 @@ def bulk_update_fields(model_class, update_list, fields_to_update, batch_size=10
             model_class.objects.bulk_update(page.object_list, fields_to_update)
 
 
+class ScriptTemplate(BaseCommand):
+    """
+    Contains common script actions for our scripts which can be prefilled as templates.
+    """
+
+    def mass_populate_field(self, sender, filter_conditions, fields_to_update):
+        """Loops through each valid "sender" object - specified by filter_conditions - and
+        updates fields defined by fields_to_update using populate_function.
+
+        You must define populate_field before you can use this function.
+        """
+
+        objects = sender.objects.filter(**filter_conditions)
+
+        # Code execution will stop here if the user prompts "N"
+        TerminalHelper.prompt_for_execution(
+            system_exit_on_terminate=True,
+            info_to_inspect=f"""
+            ==Proposed Changes==
+            Number of {sender} objects to change: {len(objects)}
+            These fields will be updated on each record: {fields_to_update}
+            """,
+            prompt_title="Do you wish to patch this data?",
+        )
+        logger.info("Updating...")
+
+        to_update: List[sender] = []
+        failed_to_update: List[sender] = []
+        for updated_object in objects:
+            try:
+                self.populate_field(updated_object)
```

Comment:
Noticed that populate_field(self, field_to_update) always raises a NotImplementedError and was wondering what the reasoning was behind doing a try/catch for a function that will always raise an exception